### DISCUSSION TITLE
How to uninstall Tinygo on Windows? And how to update it?

### DISCUSSION THREAD

**Discussion by Gustavomurta** [tinygo-org/tinygo#4656] [Q&A]
**My current environment is:**
Windows 10 
go version go1.23.4 windows/amd64
tinygo version 0.32.0 windows/amd64 (using go version go1.23.4 and LLVM version 18.1.2)

**How to update Tinygo to current version - 0.34.0?**

When I try to update through Scoop, I get this message.

C:\Users\jgust>scoop update tinygo
**tinygo: 0.32.0 (latest version)**
Latest versions for all apps are installed! For more information try 'scoop status'

I downloaded the current version from the Github website:
_tinygo0.34.0.windows-amd64.zip_

But I think I have to uninstall version 0.32.0 first, right?
But how do I uninstall the version installed on Windows?
I did not find instructions for this procedure.

Directory where Tinygo is installed:
C:\Users\jgust\scoop\apps\tinygo\0.32.0

**Searching the Scoop WIKI:**
**C:\Users\jgust>scoop uninstall tinygo  (I suggest this command in the Installation tutorial)****
Uninstalling 'tinygo' (0.32.0).
Removing shim 'tinygo.shim'.
Removing shim 'tinygo.exe'.
Unlinking ~\scoop\\apps\tinygo\current
'tinygo' was uninstalled.

****C:\Users\jgust>scoop install tinygo** 
Installing 'tinygo' (0.32.0) [64bit] from 'main' bucket
Loading tinygo0.32.0.windows-amd64.zip from cache
Checking hash of tinygo0.32.0.windows-amd64.zip ... ok.
Extracting tinygo0.32.0.windows-amd64.zip ... done.
Linking ~\scoop\apps\tinygo\current => ~\scoop\apps\tinygo\0.32.0
Creating shim for 'tinygo'.
'tinygo' (0.32.0) was installed successfully!
'tinygo' suggests installing 'go'.

C:\Users\jgust>

**Conclusion - I was unable to update Tinygo on Windows, using the scoop.**

**Comment by aykevl** [comment]
I think you need to run `scoop update` to fetch the latest package versions.

I checked, and TinyGo is up to date in Scoop: https://github.com/ScoopInstaller/Main/blob/master/bucket/tinygo.json

**Reply by Gustavomurta** [reply]
Thank you very much Ayke for your quick response.
My scoop was having problems.

I had to remove and reinstall the scoop.
https://github.com/ScoopInstaller/Scoop/wiki/Uninstalling-Scoop
https://github.com/ScoopInstaller/Scoop/wiki/Quick-Start

Now everything is correct!


```
C:\Users\jgust>scoop install tinygo
Installing 'tinygo' (0.34.0) [64bit] from 'main' bucket
tinygo0.34.0.windows-amd64.zip (127,9 MB) [===================================================================] 100%
Checking hash of tinygo0.34.0.windows-amd64.zip ... ok.
Extracting tinygo0.34.0.windows-amd64.zip ... done.
Linking ~\scoop\apps\tinygo\current => ~\scoop\apps\tinygo\0.34.0
Creating shim for 'tinygo'.
'tinygo' (0.34.0) was installed successfully!
'tinygo' suggests installing 'go'.

C:\Users\jgust>
```

**Comment by Gustavomurta** [comment]
Now everything is correct!
